src/main/java/org/apache/sling/commons/threads/impl/ThreadLocalCleaner.java

```diff
@@ -159,6 +164,9 @@ public void cleanup() {
             diff(inheritableThreadLocalsField, inheritableThreadLocalsCopy.references);
         }
         restoreOldThreadLocals();
+        if(cleanupLogCache.add(Thread.currentThread().getId())) {
+            LOG.warn("Cleanup of thread locals performed for {}", Thread.currentThread());
```

Comment:
The WARN message is not helpful, because it requires a mapping of these IDs (which are unpredictable by nature) to some (misbehaving) code to be helpful. But such a mapping does not exist.

I suggest that we rather log the name of the cleaned threadlocal variable, which then can be used to search the codebase, plus the threadpool in which this thread was running.